synapse/api/auth/msc3861_delegated.py

```diff
@@ -226,6 +226,7 @@ async def get_user_by_req(
             # TODO: we probably want to assert the allow_guest inside this call
             # so that we don't provision the user if they don't have enough permission:
             requester = await self.get_user_by_access_token(access_token, allow_expired)
+            await self._record_request(request, requester)
```

Comment:
I think this should be outside the if so that it also gets tracked for AS users?

Reply:
Ugh, this was deliberate in that I _thought_ that's what the old code did. But it doesn't: https://github.com/matrix-org/synapse/blob/962e1790d0789c1e6f292d8b55a5b0db2afd41a7/synapse/api/auth/internal.py#L123-L188

Reply:
38e3fe7 should fix.